components/core/src/clp_s/ArchiveWriter.hpp

```diff
@@ -31,6 +33,49 @@ struct ArchiveWriterOption {
     std::string authoritative_timestamp_namespace;
 };
 
+struct ArchiveStats {
+    explicit ArchiveStats(
+            std::string id,
+            epochtime_t begin_timestamp,
+            epochtime_t end_timestamp,
+            size_t uncompressed_size,
+            size_t compressed_size,
+            nlohmann::json range_index,
+            bool is_split
+    )
+            : id{id},
+              begin_timestamp{begin_timestamp},
+              end_timestamp{end_timestamp},
+              uncompressed_size{uncompressed_size},
+              compressed_size{compressed_size},
+              range_index(std::move(range_index)),  // Avoid {} to prevent wrapping in JSON array.
+              is_split{is_split} {}
+
+    [[nodiscard]] auto as_string() const -> std::string {
+        namespace Archive = clp::streaming_archive::cMetadataDB::Archive;
+        namespace File = clp::streaming_archive::cMetadataDB::File;
+        constexpr std::string_view cRangeIndex{"range_index"};
+
+        nlohmann::json json_msg
+                = {{Archive::Id, id},
+                   {Archive::BeginTimestamp, begin_timestamp},
+                   {Archive::EndTimestamp, end_timestamp},
+                   {Archive::UncompressedSize, uncompressed_size},
+                   {Archive::Size, compressed_size},
+                   {File::IsSplit, is_split},
+                   {cRangeIndex, range_index}};
+        return json_msg.dump(-1, ' ', false, nlohmann::json::error_handler_t::ignore);
+    }
+
+    std::string const id;
+    epochtime_t const begin_timestamp{};
+    epochtime_t const end_timestamp{};
+    size_t const uncompressed_size{};
+    size_t const compressed_size{};
+    nlohmann::json const range_index;
+    bool const is_split{};
```

Comment:
I guess the reason you're const qualifying these variables is so that they're not independently modified by a caller? clang-tidy understandably has concerns about const-qualified members, so I wonder if we should just convert `ArchiveStats` to a full-fledged class so that these members are private? Is any caller actually using these members directly?